### PR TITLE
router: adds a handler to add request id header

### PR DESCRIPTION
**Pull request by andrestc**
This commit adds a new Handler, near the end of the pipeline, that
adds a UUIDv4 header to the request (if the header is already not set).
This behavior is controlled thru an environment variable, that must be
set to the header name.

### CODE REVIEW

**Comment by tibraga** [comment]
@andrestc Curiosity. For what purpose?

**Comment by andrestc** [comment]
hey @tibraga, we talked to @tuxmonteiro about it, the next step will be to add this request ID to the access log and have a way to trace the entire lifetime of the request between multiple services

File: router/src/main/java/io/galeb/router/handlers/PoolHandler.java [hunk 1]
```diff
@@ -49,17 +49,23 @@
...
     private ProxyHandler proxyHandler = null;
+    private RequestIDHandler requestIDHandler = null;
     private ExtendedLoadBalancingProxyClient proxyClient;
 
     private final Pool pool;
 
     public PoolHandler(final Pool pool) {
         this.pool = pool;
         this.defaultHandler = buildPoolHandler();
+        if (!requestIDHeader.equals("")) {
```
Comment (tuxmonteiro) [inline router/src/main/java/io/galeb/router/handlers/PoolHandler.java]: @andrestc I recommend change to "".equals(requestIDHeader) 
Best pratices prevents NullPointerException. If you use variable.equals("xx"), then you will get a NullPointerException if variable is null. Putting the constant string on the left hand side avoids this possibility.

File: router/src/main/java/io/galeb/router/handlers/RequestIDHandler.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+public class RequestIDHandler implements HttpHandler {
+
+    private final HttpString requestIDHeader;
+
+    public RequestIDHandler(final String requestIDHeader) {
+        this.requestIDHeader = new HttpString(requestIDHeader);
+    }
+
+    @Override
+    public void handleRequest(HttpServerExchange exchange) throws Exception {
```
Comment (tuxmonteiro) [inline router/src/main/java/io/galeb/router/handlers/RequestIDHandler.java]: @andrestc The handleRequest needs call the next handler (in this case, the proxyHandler (if not null) ou defaultHandler.
I recommend add constructor param with the "next" handler.

**Comment by tuxmonteiro** [comment]
@andrestc See the comments added in code. Otherwise, we thanks.

**Comment by tuxmonteiro** [comment]
Moved to https://github.com/galeb/galeb4/pull/51

**Pull request closed without merging**